src/core/truthTable/truth_table.cpp

```diff
@@ -29,38 +29,45 @@ namespace syrec {
         return result;
     }
 
-    auto TruthTable::equal(TruthTable& tt1, TruthTable& tt2, bool equalityUpToDontCare) -> bool {
+    auto TruthTable::equal(TruthTable const& tt1, TruthTable const& tt2, bool equalityUpToDontCare) -> bool {
         if (!equalityUpToDontCare) {
             return (tt1 == tt2);
         }
 
-        if (tt1.nInputs() != tt2.nInputs()) {
+        // both the truth table must have the same number of I/O combinations.
+        if (tt1.size() != tt2.size()) {
```

Comment:
must the TTs really have the same size?
IMHO, they must only have the same number of primary inputs and primary outputs. If one of the TTs contains constants, while the other does not, then the TTs will not be considered equal.